contracts/src/libraries/LPMath.sol

```diff
@@ -887,15 +906,25 @@ library LPMath {
 
             // NOTE: Round the delta down to avoid overshooting.
             //
+            // Calculate the objective function's value. If the value's magnitude
+            // is smaller than the previous smallest value, then we update the
+            // value and record the share proceeds. We'll ultimately return the
+            // share proceeds that resulted in the smallest value.
+            int256 delta = presentValue.mulDown(lpTotalSupply).toInt256() -
+                params
+                    .startingPresentValue
+                    .mulUp(params.activeLpTotalSupply)
+                    .toInt256();
+            if (smallestDelta == 0 || delta.abs() < smallestDelta.abs()) {
+                smallestDelta = delta;
+                closestShareProceeds = shareProceeds;
+                closestPresentValue = presentValue;
+            }
```

Comment:
just adding my explanation here: this updates the solution candidate whenever `|F(dz)|` gets smaller, meaning `F(dz)` gets closer to `0`, the exact solution.